When restarting my application, I get MolgenisValidationException: column this.labelen#d3b9fc1f does not exist

### How to Reproduce
- Start MOLGENIS with empty database
- Restart you application

### Observed behaviour
```
Caused by: org.molgenis.data.validation.MolgenisValidationException: column this.labelen#d3b9fc1f does not exist
	at org.molgenis.data.postgresql.PostgreSqlExceptionTranslator.translateUndefinedColumnException(PostgreSqlExceptionTranslator.java:400) ~[molgenis-data-postgresql-2.1.0-SNAPSHOT.jar:na]
	at org.molgenis.data.postgresql.PostgreSqlExceptionTranslator.doTranslate(PostgreSqlExceptionTranslator.java:114) ~[molgenis-data-postgresql-2.1.0-SNAPSHOT.jar:na]
	at org.molgenis.data.postgresql.PostgreSqlExceptionTranslator.doTranslate(PostgreSqlExceptionTranslator.java:88) ~[molgenis-data-postgresql-2.1.0-SNAPSHOT.jar:na]
	at org.molgenis.data.postgresql.PostgreSqlExceptionTranslator.doTranslate(PostgreSqlExceptionTranslator.java:53) ~[molgenis-data-postgresql-2.1.0-SNAPSHOT.jar:na]
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73) ~[spring-jdbc-4.3.4.RELEASE.jar:4.3.4.RELEASE]
```

--- a/molgenis-data-i18n/src/main/java/org/molgenis/data/i18n/SystemEntityTypeI18nInitializer.java
+++ b/molgenis-data-i18n/src/main/java/org/molgenis/data/i18n/SystemEntityTypeI18nInitializer.java
@@ -1,22 +1,18 @@
 package org.molgenis.data.i18n;
 
-import org.molgenis.data.DataService;
 import org.molgenis.data.i18n.model.I18nStringMetaData;
-import org.molgenis.data.i18n.model.LanguageFactory;
-import org.molgenis.data.meta.MetaDataService;
 import org.molgenis.data.meta.SystemEntityType;
 import org.molgenis.data.meta.model.AttributeMetadata;
 import org.molgenis.data.meta.model.EntityTypeMetadata;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.event.ContextRefreshedEvent;
 import org.springframework.stereotype.Component;
 
 import java.util.stream.Stream;
 
-import static java.util.Objects.requireNonNull;
 import static org.molgenis.data.meta.AttributeType.STRING;
 import static org.molgenis.data.meta.AttributeType.TEXT;
+import static org.molgenis.data.support.AttributeUtils.getI18nAttributeName;
 
 /**
  * Enriches system entity meta data with internationalization attributes.
@@ -26,19 +22,6 @@ import static org.molgenis.data.meta.AttributeType.TEXT;
 @Component
 public class SystemEntityTypeI18nInitializer
 {
-	private final MetaDataService metaDataService;
-	private final DataService dataService;
-	private final LanguageFactory languageFactory;
-
-	@Autowired
-	public SystemEntityTypeI18nInitializer(MetaDataService metaDataService, DataService dataService,
-			LanguageFactory languageFactory)
-	{
-		this.metaDataService = requireNonNull(metaDataService);
-		this.dataService = requireNonNull(dataService);
-		this.languageFactory = requireNonNull(languageFactory);
-	}
-
 	/**
 	 * Initialize internationalization attributes
 	 *
@@ -55,14 +38,14 @@ public class SystemEntityTypeI18nInitializer
 
 		languageCodes.forEach(languageCode ->
 		{
-			entityTypeMeta.addAttribute(EntityTypeMetadata.LABEL + '-' + languageCode).setNillable(true)
+			entityTypeMeta.addAttribute(getI18nAttributeName(EntityTypeMetadata.LABEL, languageCode)).setNillable(true)
 					.setLabel("Label (" + languageCode + ')');
-			entityTypeMeta.addAttribute(EntityTypeMetadata.DESCRIPTION + '-' + languageCode).setNillable(true)
-					.setLabel("Description (" + languageCode + ')').setDataType(TEXT);
-			attrMetaMeta.addAttribute(AttributeMetadata.LABEL + '-' + languageCode).setNillable(true)
+			entityTypeMeta.addAttribute(getI18nAttributeName(EntityTypeMetadata.DESCRIPTION, languageCode))
+					.setNillable(true).setLabel("Description (" + languageCode + ')').setDataType(TEXT);
+			attrMetaMeta.addAttribute(getI18nAttributeName(AttributeMetadata.LABEL, languageCode)).setNillable(true)
 					.setLabel("Label (" + languageCode + ')');
-			attrMetaMeta.addAttribute(AttributeMetadata.DESCRIPTION + '-' + languageCode).setNillable(true)
-					.setLabel("Description (" + languageCode + ')').setDataType(TEXT);
+			attrMetaMeta.addAttribute(getI18nAttributeName(AttributeMetadata.DESCRIPTION, languageCode))
+					.setNillable(true).setLabel("Description (" + languageCode + ')').setDataType(TEXT);
 			i18nStringMeta.addAttribute(languageCode).setNillable(true).setDataType(STRING);
 		});
 	}

--- a/molgenis-data-rest/src/test/java/org/molgenis/data/rest/v2/RestControllerV2Test.java
+++ b/molgenis-data-rest/src/test/java/org/molgenis/data/rest/v2/RestControllerV2Test.java
@@ -5,8 +5,6 @@ import com.google.gson.Gson;
 import com.google.gson.JsonSyntaxException;
 import com.google.gson.reflect.TypeToken;
 import org.mockito.Matchers;
-import org.mockito.invocation.InvocationOnMock;
-import org.mockito.stubbing.Answer;
 import org.molgenis.data.*;
 import org.molgenis.data.i18n.LanguageService;
 import org.molgenis.data.meta.AttributeType;
@@ -62,6 +60,7 @@ import static org.mockito.Matchers.any;
 import static org.mockito.Matchers.eq;
 import static org.mockito.Mockito.*;
 import static org.molgenis.data.EntityManager.CreationMode.POPULATE;
+import static org.molgenis.data.i18n.LanguageService.DEFAULT_LANGUAGE_CODE;
 import static org.molgenis.data.meta.AttributeType.*;
 import static org.molgenis.data.meta.model.EntityType.AttributeRole.*;
 import static org.molgenis.data.meta.model.Package.PACKAGE_SEPARATOR;
@@ -101,6 +100,9 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 	@Autowired
 	private AttributeFactory attributeFactory;
 
+	@Autowired
+	private LanguageService languageService;
+	
 	@Autowired
 	private RestControllerV2 restControllerV2;
 
@@ -349,15 +351,9 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 		assertEquals(refRefEntity.getIdValue(), REF_REF_ENTITY_ID);
 		assertEquals(selfRefEntity.getIdValue(), "0");
 
-		when(entityManager.create(entityType, POPULATE)).thenAnswer(new Answer<Entity>()
-		{
-			@Override
-			public Entity answer(InvocationOnMock invocation) throws Throwable
-			{
-				return new DynamicEntity(entityType);
-			}
-		});
+		when(entityManager.create(entityType, POPULATE)).thenAnswer(invocation -> new DynamicEntity(entityType));
 
+		when(languageService.getCurrentUserLanguageCode()).thenReturn(DEFAULT_LANGUAGE_CODE);
 		mockMvc = MockMvcBuilders.standaloneSetup(restControllerV2).setMessageConverters(gsonHttpMessageConverter)
 				.setConversionService(conversionService).build();
 	}
@@ -1049,7 +1045,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": false,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1069,7 +1066,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": false,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1187,7 +1185,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": false,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1224,7 +1223,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": false,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1250,7 +1250,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": true,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1270,7 +1271,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": true,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1348,7 +1350,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": true,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1387,13 +1390,15 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 			+ "              \"lookupAttribute\": true,\n" + "              \"isAggregatable\": false\n"
 			+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 			+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n" + "            \"id\"\n"
-			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false\n"
+			+ "          ],\n" + "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+			+ "          \"languageCode\": \"en\"\n"
 			+ "        },\n" + "        \"auto\": false,\n" + "        \"nillable\": true,\n"
 			+ "        \"readOnly\": false,\n" + "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 			+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
 			+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n" + "    \"labelAttribute\": \"id\",\n"
 			+ "    \"idAttribute\": \"id\",\n" + "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n"
-			+ "    \"isAbstract\": false,\n" + "    \"writable\": false\n" + "  },\n"
+			+ "    \"isAbstract\": false,\n" + "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n"
+			+ "  },\n"
 			+ "  \"_href\": \"/api/v2/entity/0\",\n" + "  \"id\": \"0\",\n" + "  \"bool\": true,\n"
 			+ "  \"categorical\": {\n" + "    \"_href\": \"/api/v2/refEntity/ref0\",\n" + "    \"id\": \"ref0\"\n"
 			+ "  },\n" + "  \"categorical_mref\": [\n" + "    {\n" + "      \"_href\": \"/api/v2/refEntity/ref0\",\n"
@@ -1423,7 +1428,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "        \"lookupAttribute\": false,\n" + "        \"isAggregatable\": false\n" + "      }\n"
 					+ "    ],\n" + "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"bool\": true\n" + "}";
 
 	private final String resourcePartialAttributeInCompoundResponse =
@@ -1447,7 +1453,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"compound_attr0\": \"compoundAttr0Str\"\n" + "}";
 
 	private final String resourcePartialAttributeInCompoundInCompoundResponse =
@@ -1482,7 +1489,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"compound_attrcompound_attr0\": \"compoundAttrCompoundAttr0Str\"\n" + "}";
 
 	private final String resourcePartialSubAttributeResponse =
@@ -1505,14 +1513,16 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"xref\": {\n" + "    \"_href\": \"/api/v2/refEntity/ref0\",\n" + "    \"value\": \"val0\"\n"
 					+ "  }\n" + "}";
 
@@ -1544,14 +1554,16 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"xref\": {\n" + "    \"_href\": \"/api/v2/refEntity/ref0\",\n" + "    \"id\": \"ref0\",\n"
 					+ "    \"value\": \"val0\"\n" + "  }\n" + "}";
 
@@ -1574,7 +1586,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "        \"lookupAttribute\": false,\n" + "        \"isAggregatable\": false\n" + "      }\n"
 					+ "    ],\n" + "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"bool\": true,\n" + "  \"string\": \"str\"\n" + "}";
 
 	private final String resourceCollectionResponse =
@@ -1610,7 +1623,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1631,7 +1645,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1766,7 +1781,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1808,7 +1824,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1835,7 +1852,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": true,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1857,7 +1875,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": true,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1941,7 +1960,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": true,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
@@ -1983,14 +2003,16 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "              \"isAggregatable\": false\n" + "            }\n" + "          ],\n"
 					+ "          \"labelAttribute\": \"id\",\n" + "          \"idAttribute\": \"id\",\n"
 					+ "          \"lookupAttributes\": [\n" + "            \"id\"\n" + "          ],\n"
-					+ "          \"isAbstract\": false,\n" + "          \"writable\": false\n" + "        },\n"
+					+ "          \"isAbstract\": false,\n" + "          \"writable\": false,\n"
+					+ "          \"languageCode\": \"en\"\n" + "        },\n"
 					+ "        \"auto\": false,\n" + "        \"nillable\": true,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"start\": 0,\n" + "  \"num\": 100,\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n" + "  \"start\": 0,\n"
+					+ "  \"num\": 100,\n"
 					+ "  \"total\": 2,\n" + "  \"items\": [\n" + "    {\n" + "      \"_href\": \"/api/v2/entity/0\",\n"
 					+ "      \"id\": \"0\",\n" + "      \"bool\": true,\n" + "      \"categorical\": {\n"
 					+ "        \"_href\": \"/api/v2/refEntity/ref0\",\n" + "        \"id\": \"ref0\"\n" + "      },\n"
@@ -2050,7 +2072,8 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "                ],\n" + "                \"labelAttribute\": \"id\",\n"
 					+ "                \"idAttribute\": \"id\",\n" + "                \"lookupAttributes\": [\n"
 					+ "                  \"id\"\n" + "                ],\n" + "                \"isAbstract\": false,\n"
-					+ "                \"writable\": false\n" + "              },\n"
+					+ "                \"writable\": false,\n" + "                \"languageCode\": \"en\"\n"
+					+ "              },\n"
 					+ "              \"auto\": false,\n" + "              \"nillable\": true,\n"
 					+ "              \"readOnly\": false,\n" + "              \"labelAttribute\": false,\n"
 					+ "              \"unique\": false,\n" + "              \"visible\": true,\n"
@@ -2058,14 +2081,16 @@ public class RestControllerV2Test extends AbstractMolgenisSpringTest
 					+ "            }\n" + "          ],\n" + "          \"labelAttribute\": \"id\",\n"
 					+ "          \"idAttribute\": \"id\",\n" + "          \"lookupAttributes\": [\n"
 					+ "            \"id\"\n" + "          ],\n" + "          \"isAbstract\": false,\n"
-					+ "          \"writable\": false\n" + "        },\n" + "        \"auto\": false,\n"
+					+ "          \"writable\": false,\n" + "          \"languageCode\": \"en\"\n" + "        },\n"
+					+ "        \"auto\": false,\n"
 					+ "        \"nillable\": false,\n" + "        \"readOnly\": false,\n"
 					+ "        \"labelAttribute\": false,\n" + "        \"unique\": false,\n"
 					+ "        \"visible\": true,\n" + "        \"lookupAttribute\": false,\n"
 					+ "        \"isAggregatable\": false\n" + "      }\n" + "    ],\n"
 					+ "    \"labelAttribute\": \"id\",\n" + "    \"idAttribute\": \"id\",\n"
 					+ "    \"lookupAttributes\": [\n" + "      \"id\"\n" + "    ],\n" + "    \"isAbstract\": false,\n"
-					+ "    \"writable\": false\n" + "  },\n" + "  \"_href\": \"/api/v2/entity/0\",\n"
+					+ "    \"writable\": false,\n" + "    \"languageCode\": \"en\"\n" + "  },\n"
+					+ "  \"_href\": \"/api/v2/entity/0\",\n"
 					+ "  \"xref\": {\n" + "    \"_href\": \"/api/v2/refEntity/ref0\",\n" + "    \"id\": \"ref0\",\n"
 					+ "    \"ref\": {\n" + "      \"_href\": \"/api/v2/refRefEntity/refRef0\",\n"
 					+ "      \"value\": \"value\"\n" + "    }\n" + "  }\n" + "}";

--- a/molgenis-data/src/main/java/org/molgenis/data/support/AttributeUtils.java
+++ b/molgenis-data/src/main/java/org/molgenis/data/support/AttributeUtils.java
@@ -6,6 +6,7 @@ import org.molgenis.data.meta.model.Attribute;
 import java.util.EnumSet;
 
 import static org.molgenis.data.meta.AttributeType.*;
+import static org.springframework.util.StringUtils.capitalize;
 
 public class AttributeUtils
 {
@@ -15,7 +16,7 @@ public class AttributeUtils
 
 	public static String getI18nAttributeName(String attrName, String languageCode)
 	{
-		return attrName + '-' + languageCode;
+		return attrName + capitalize(languageCode.toLowerCase());
 	}
 
 	/**

--- a/molgenis-data/src/test/java/org/molgenis/data/support/AttributeUtilsTest.java
+++ b/molgenis-data/src/test/java/org/molgenis/data/support/AttributeUtilsTest.java
@@ -5,8 +5,10 @@ import org.molgenis.data.meta.model.Attribute;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Iterator;
+import java.util.List;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
@@ -32,5 +34,21 @@ public class AttributeUtilsTest
 	{
 		Attribute attr = when(mock(Attribute.class).getDataType()).thenReturn(attrType).getMock();
 		assertEquals(AttributeUtils.isIdAttributeTypeAllowed(attr), validIdAttrType);
+	}
+
+	@DataProvider(name = "getI18nAttributeNameProvider")
+	public static Iterator<Object[]> getI18nAttributeNameProvider()
+	{
+		List<Object[]> dataList = new ArrayList<>();
+		dataList.add(new Object[] { "lang", "en", "langEn" });
+		dataList.add(new Object[] { "lang", "En", "langEn" });
+		dataList.add(new Object[] { "lang", "EN", "langEn" });
+		return dataList.iterator();
+	}
+
+	@Test(dataProvider = "getI18nAttributeNameProvider")
+	public void testGetI18nAttributeName(String attrName, String languageCode, String i18nAttrName)
+	{
+		assertEquals(AttributeUtils.getI18nAttributeName(attrName, languageCode), i18nAttrName);
 	}
 }

--- a/molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/platform/PlatformIT.java
+++ b/molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/platform/PlatformIT.java
@@ -234,8 +234,8 @@ public class PlatformIT extends AbstractTestNGSpringContextTests
 	@Test(singleThreaded = true)
 	public void testLanguageService()
 	{
-		assertEquals(dataService.getMeta().getEntityType(ENTITY_TYPE_META_DATA).getAttribute("label-en").getName(),
-				"label-en");
+		assertEquals(dataService.getMeta().getEntityType(ENTITY_TYPE_META_DATA).getAttribute("labelEn").getName(),
+				"labelEn");
 		assertEquals(dataService.getMeta().getEntityType(ENTITY_TYPE_META_DATA).getLabelAttribute("en").getName(),
 				"simpleName");
 		assertEquals(dataService.getMeta().getEntityType(ENTITY_TYPE_META_DATA).getLabelAttribute("pt").getName(),
@@ -251,17 +251,17 @@ public class PlatformIT extends AbstractTestNGSpringContextTests
 
 		// NL
 		assertNotNull(dataService.getEntityType(I18N_STRING).getAttribute("nl"));
-		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("label-nl"));
-		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("description-nl"));
-		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("label-nl"));
-		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("description-nl"));
+		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("labelNl"));
+		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("descriptionNl"));
+		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("labelNl"));
+		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("descriptionNl"));
 
 		// EN
 		assertNotNull(dataService.getEntityType(I18N_STRING).getAttribute("en"));
-		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("label-en"));
-		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("description-en"));
-		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("label-en"));
-		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("description-en"));
+		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("labelEn"));
+		assertNotNull(dataService.getEntityType(ENTITY_TYPE_META_DATA).getAttribute("descriptionEn"));
+		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("labelEn"));
+		assertNotNull(dataService.getEntityType(ATTRIBUTE_META_DATA).getAttribute("descriptionEn"));
 
 		Entity car = new DynamicEntity(i18nStringMetaData);
 		car.set(I18nStringMetaData.MSGID, "car");

